### DISCUSSION TITLE
Fungsi yang terdapat di duration

### DISCUSSION THREAD

**Discussion by rizkiajimahardika** [teamalgoritma/community#103] [Q&A] [answered]
izin bertanya, saya masih bingung mengenai fungsi dibawah ini
![1](https://user-images.githubusercontent.com/79896604/168325535-cebb4ccc-c5db-4e06-9862-0670756c076e.png)
![2](https://user-images.githubusercontent.com/79896604/168325554-66e6d820-ae0f-45c7-add4-872743310de3.png)
![3](https://user-images.githubusercontent.com/79896604/168325597-28bb5384-cf31-4ecb-9f27-b32b2e784f54.png)

**Comment by NabiilahArdini** [comment, marked as answer]
Halo @rizkiajimahardika 

Fungsi `substr()` adalah fungsi untuk mengekstrak karakter tertentu dari text. Pada code di atas, sepertinya penulis code ingin mengekstrak hanya **angka durasi menit** nya dari kolom movie duration.

Berdasarkan [dokumentasi fungsi](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/substr), penulisan code tersebut mengikuti penulisan code versi `substr(x, start, stop)` dimana parameternya:

- `x`: data
- `start`: kapan mulai extraksi karakter
- `stop`: kapan stop ekstraksi karakter

untuk code yang ada tanyakan, `substr(duration, 1, nchar(as.character(duration)-4)` berarti:
- x = `duration` -> ingin diambil dari kolom duration
- start = `1` -> ekstrak dari karakter pertama
- stop = `nchar(as.character(duration)-4` -> kolom durasi dijadikan karakter terlebih dulu kemudian dihitung jumlah karakternya, dan dikurangi dengan 4 (untuk menghilangkan pattern " min" yang berjumlah 4 karakter dari belakang)

Semoga informasi ini dapat membantu.
